src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -4418,19 +4442,114 @@ def generate_agg_qc(
                 )
 
             else:
-                for agg_func, cols in agg_func_to_col_map.items():
-                    col_single_agg_func_map = {
-                        column: AggFuncInfo(
-                            func=agg_func if column in cols else "min",
-                            is_dummy_agg=column not in cols,
+                if uses_named_aggs:
+                    # If this is true, then we are dealing with agg with NamedAggregations.
+
+                    def generate_single_agg_column_func_map(
+                        identifier_pair: PandasLabelToSnowflakeIdentifierPair,
+                        agg_func: AggFuncWithLabel,
+                    ) -> dict[PandasLabelToSnowflakeIdentifierPair, AggFuncInfo]:
+                        """
+                        Helper function to produce the agg func map for a single aggregation, including dummy aggregations.
+
+                        Notes:
+                        Adds dummy aggregations for all columns that will be aggregated over.
+                        """
+                        col_single_agg_func_map = {}
+                        for c in column_to_agg_func.keys():
+                            if c == identifier_pair:
```

Comment:
Why not `if identifier_pair in column_to_agg_func` + `col_single_agg_func_map[identifier_pair] = (...)`?

Reply:
I'm not sure I follow this comment?

Reply:
Never mind, I think I misunderstood the purpose of this function. I thought it was only generating a single column corresponding to `identifier_pair`.

I didn't realize that unlike df.agg w/ list or dict arguments, which combine common aggregations together, the named aggregation code path gives every single argument its own 1-hot row.

```
>>> df = pd.DataFrame([[1, 2, 3],
                   [4, 5, 6],
                   [7, 8, 9],
                   [np.nan, np.nan, np.nan]],
                  columns=['A', 'B', 'C'])
>>> df.agg(x=("A", min), y=("B", max), z=("A", max))
     A    B
x  1.0  NaN
y  NaN  8.0
z  7.0  NaN
>>> df.agg({"A": [min, max], "B": max})
       A    B
min  1.0  NaN
max  7.0  8.0
```